File: javascript/scripts/sync-version.mjs
#!/usr/bin/env node
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const packageRoot = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..");
const pkg = JSON.parse(fs.readFileSync(path.join(packageRoot, "package.json"), "utf8"));
const version = pkg.version;
if (!/^\d+\.\d+\.\d+(-[\w.-]+)?$/.test(version)) {
  console.error(`Invalid version in package.json: ${version}`);
  process.exit(1);
}

const versionTs = `export const VERSION = "${version}";\n\nexport const USER_AGENT = \`@agent-api/sdk/\${VERSION}\`;\n`;
fs.writeFileSync(path.join(packageRoot, "src", "version.ts"), versionTs);
console.log(`Synced @agent-api/sdk version ${version} -> src/version.ts`);

File: python/scripts/sync_version.py
#!/usr/bin/env python3
from __future__ import annotations

import pathlib
import re
import sys

ROOT = pathlib.Path(__file__).resolve().parents[1]
PYPROJECT = ROOT / "pyproject.toml"
VERSION_PY = ROOT / "src" / "agent_api" / "_version.py"

match = re.search(r'^version = "(.+)"$', PYPROJECT.read_text(), re.MULTILINE)
if not match:
    print("Could not read version from pyproject.toml", file=sys.stderr)
    sys.exit(1)

version = match.group(1)
VERSION_PY.write_text(
    f'''"""Agent API Python SDK version."""

__version__ = "{version}"
USER_AGENT = f"cloudsway-agent/{{__version__}}"

DEFAULT_TIMEOUT = 600.0
DEFAULT_STREAM_TIMEOUT = 3600.0
DEFAULT_MAX_RETRIES = 2
'''
)
print(f"Synced cloudsway-agent version {version} -> src/agent_api/_version.py")
